File: python/1 - Two Sum.py
# 1
# Two Sum
# https://leetcode.com/problems/two-sum/
# python
# easy
# O(n)
# O(n)
# basic

# solution can be imporved for time and space
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i,n in enumerate(nums):
            if target-n in d:
                return [d[target-n], i]
            d[n]=i
        return False

File: java/1 - Two Sum.java
// 1
// Two Sum
// https://leetcode.com/problems/two-sum/
// java
// easy
// O(n^2)
// O(1)
// basic

// O(n^2) time BRUTE FORCE METHOD VERY SLOW (see c++ or python implementation for O(n))
class Solution {
    public int[] twoSum(int[] nums, int target) {
        for (int i = 0; i < nums.length-1; i++) {
            for (int j = i+1; j < nums.length; j++) {
                if (nums[i] + nums[j] == target) {
                    return new int[] {i, j};
                }
            }
        }
        
        return null;
    }
}
